packages/blade/src/components/TextArea/_decisions/decisions.md

```diff
@@ -0,0 +1,66 @@
+| name                 | `string`                         |         | The name of the input, useful in form submissions                                                                                                                                                                                                            |          |
+| onChange             | `({ event, name, value }) => {}` |         | Function called when the value of the input field changes                                                                                                                                                                                                    |          |
+| onBlur               | `({ event, name, value }) => {}` |         | Function called when the input loses focus                                                                                                                                                                                                                   |          |
+| isDisabled           | `boolean`                        | `false` | Marks the input field as disabled. It'll also set aria-disabled to true and hence the input field will loose keyboard focus                                                                                                                                  |          |
+| isRequired           | `boolean`                        | `false` | Marks the input field as required. if the errorText is provided then that will be shown if the field is required and not filled, else ${label} is required                                                                                                   |          |
+| neccessityIndicator  | `required, optional`             |         | Indicator to be shown next to the label for emphasizing the necessity of the field. if optional is provided then it'll show `(optional)` and if required is provided then it'll show `*`                                                                     |          |
+| showClearButton      | `boolean`                        | `true`  | Defines if we want to show the clear icon button on the right side of the password field                                                                                                                                                                     |          |
+| onClearButtonClicked | `() => {}`                       |         | Used in conjunction with showClearButton prop. If clicked it removes the contents of the TextArea. Users can also hook their own analytics code or do something else.                                                                                        |          |
```

Comment:
How about `onClickClearButon`?

Reply:
Seems kinda reversed 😅 
In BaseText input we also have the same name. OnClearButtonClicked.

Reply:
just `onClear` sounds okay to me too though. Bit shorter.

Reply:
Oh, I think we should have the same convention in that case. Would favor `onClick...` because we follow present tense at some places like `onDismiss` in `Alert` and `onClick` in `Button`

Reply:
Or we need to think about how should we approach such props naming convention 🤔

Reply:
don't mind changing in BaseInput to `OnClearButtonClick`. it didn't click me at that time 😅 . `onClickClearButton` should also work.